### PR TITLE
Automating TestCase 833515

### PR DESCRIPTION
**Pull request by kkrishnareddy**
Automating TestCase 833515:
PR Link for NavigateTo in google chrome: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1139

### CODE REVIEW

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                .build();
+
+        AadPromptHandler aadPromptHandler = new AadPromptHandler(promptHandlerParameters);
+        aadPromptHandler.handlePrompt(username, password);
+
+        latch.await();
+
+        //launching azure sample app and confirming user signed in or not.
+        final AzureSampleApp azureSampleApp = new AzureSampleApp();
+        azureSampleApp.install();
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:111]: will need to uninstall prior to install

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        azureSampleApp.launch();
+        Thread.sleep(TimeUnit.SECONDS.toMillis(5));
+        azureSampleApp.confirmSignedIn(username);
+
+        //signing in chrome browser.
+        IBrowser chrome = new BrowserChrome();
+        chrome.launch();
+        chrome.handleFirstRun();
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:119]: You don't need to launch and handle first run just to clear storage....you can just call clear directly.....and you already handle launch first run below so you don't need to do it twice

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        chrome.clear();
+
+        //relaunching chrome after clearing history of chrome.
+        chrome.launch();
+        chrome.handleFirstRun();
+        chrome.navigateTo("https://myapps.microsoft.com");
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:125]: navigateTo is not working....I see this method is TODO....do you have separate PR in common to implement this? Can you link these PRs?

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        UiAutomatorUtils.handleInput("i0116", username);
+        UiObject button = UiAutomatorUtils.obtainUiObjectWithText("Next");
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:127]: Let's use specific names like `nextButton`

Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:127]: Let's use specific names like nextButton

```suggestion
        UiObject nextButton = UiAutomatorUtils.obtainUiObjectWithText("Next");
```

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+        button.click();
+        UiAutomatorUtils.handleInput("i0118", password);
+        UiAutomatorUtils.handleButtonClick("idSIButton9");
+        final UiObject saveAccount = UiAutomatorUtils.obtainUiObjectWithText("No");
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:131]: This name is confusing....it saves account but the text on it seems to be `No`? Let's use more appropriate variable names

**Review by shahzaibj — CHANGES_REQUESTED** [review]
Left some comments

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java (hunk 4 above)
Comment (kkrishnareddy) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java]: changed button to nextButton

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java (hunk 5 above)
Comment (kkrishnareddy) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java]: I change to noButton. I hope it is an appropriate name.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java (hunk 3 above)
Comment (kkrishnareddy) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java]: i added new function named navigateTo(string url);

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java (hunk 2 above)
Comment (kkrishnareddy) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:122]: whenever I cleared the history and launched the chrome app it is opening from accepting terms and conditions that's why I added chrome.handleFirstRun() method.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java (hunk 1 above)
Comment (kkrishnareddy) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:114]: added azureSampleapp.uninstall();

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java (hunk 3 above)
Comment (kkrishnareddy) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java]: I have implemented the NavigatTo function in Browser Chrome and the link was provided in the description.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java (hunk 2 above)
Comment (shahzaibj) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:123]: As discussed the other day....doing the launch and handle first run should not be needed prior to doing the clear

**Comment by shahzaibj** [comment]
Let's make variables `final` wherever possible

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+// End My Shift - In Shared device mode, global sign out should work.
+// https://identitydivision.visualstudio.com/DevEx/_workitems/edit/833515
+@SupportedBrokers(brokers = {BrokerMicrosoftAuthenticator.class, BrokerHost.class})
+public class TestCase833515 extends AbstractMsalBrokerTest {
+
+    @Test
+    public void test_833515() throws MsalException, InterruptedException, UiObjectNotFoundException {
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java:62]: Can we avoid throwing the `MsalException` and `UiObjectNotFoundException` directly from the test? and rather just handle them within the test and throw it from the test by wrapping it into an `AssertionError`

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java (hunk 2 above)
Comment (kkrishnareddy) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833515.java]: done.

**Review by shahzaibj — APPROVED** [review]
